### DISCUSSION TITLE
想问一下Rule和Script有没有allforone类似的？

### DISCUSSION THREAD

**Discussion by KuHai-A** [blackmatrix7/ios_rule_script#278] [Q&A]
尤其是Rule，实在是太多了。

另外Rule里面的分类要怎么用？有的Rule在多个分类里面，比如“去广告”就在“Advertising”和“Reject”至少两个里面，如果按照这些分类去设置Surge的分组，就会出现重复了。

**Comment by KuHai-A** [comment]
还有一个问题是为什么Surge/AdvertisingScript中的跟对应的script下的要多？
比如Surge/AdvertisingScript中关于知乎的就比script/zhihu/中Plus版还要多，要以哪个为最新版啊？

**Reply by blackmatrix7** [reply]
AdvertisingScript是由https://github.com/blackmatrix7/ios_rule_script/blob/master/script/zhihu/zhihu_plus.lnplugin 自动生成的，不会有不一致的情况。

只是处理方式不同而已。

**Reply by KuHai-A** [reply]
用对比工具比对了一下，终于知道差异是怎么来的了。
ios_rule_script/rewrite/Surge/AdvertisingScript/AdvertisingScript.sgmodule下的某些条目在[Rule]和[URL Rewrite]/[Map Local]中重复了：比如“^https?:\/\/www\.zhihu\.com\/api\/v\d+\/brand\/question/\d+/card\?”这一条正则。


另外“ios_rule_script/rule/Surge/Download/”含有URL-REGEX类型，README里面写到有个mitm模块，但是此文件夹下实际并没有，是遗漏了还是就是不需要呢？

**Reply by blackmatrix7** [reply]
只有HTTPS的请求才需要MITM，对于HTTP的请求并不需要。MITM模块的主机名是通过程序推导生成的，只是起辅助作用，如果没有推导出主机名，就不会有MITM模块。

最终是否需要MITM，取决URL-REGEX的请求是否是HTTPS，而不取决于是否生成MITM模块。

**Reply by blackmatrix7** [reply]
程序推导的主机名，只是起一个辅助和参考，最终URL-REGEX类型的规则是否能生效，还是取决于是否为HTTPS。

**Reply by KuHai-A** [reply]
明白了，多谢大佬耐心指导。

咱这Rule分流有没有个整理好的大一统文件啊，难道大家都是一个一个手动加到自己配置里的吗，感觉有几百个

**Reply by LucaLin233** [reply]
> 明白了，多谢大佬耐心指导。
> 
> 
> 
> 咱这Rule分流有没有个整理好的大一统文件啊，难道大家都是一个一个手动加到自己配置里的吗，感觉有几百个

按需添加即可，没必要全加
